GitCommands/RevisionReader.cs

```diff
@@ -407,7 +407,32 @@ DateTime ParseUnixDateTime()
             // NOTE the convention is that the Subject string is duplicated at the start of the Body string
             // Therefore we read the subject twice.
             // If there are not enough characters remaining for a body, then just assign the subject string directly.
-            var body = reader.Remaining - subject.Length == 2 ? subject : reader.ReadToEnd();
+            string body;
+            string additionalData = null;
+            if (reader.Remaining - subject.Length == 2/*newlines*/ + EndOfBody.Length)
+            {
+                body = subject;
+            }
+            else
+            {
+                string tail = reader.ReadToEnd();
+                int indexOfEndOfBody = tail.LastIndexOf(EndOfBody, StringComparison.InvariantCulture);
+                if (indexOfEndOfBody < 0)
+                {
+                    // TODO log this parse error
+                    Debug.Fail("Missing end-of-body marker in the log -- this should not happen");
+                    revision = default;
+                    return false;
+                }
+
+                var containsAdditionalData = tail.Length > indexOfEndOfBody + EndOfBody.Length;
+                if (containsAdditionalData)
+                {
+                    additionalData = tail.Substring(indexOfEndOfBody + EndOfBody.Length).TrimStart();
+                }
+
+                body = indexOfEndOfBody == subject.Length + 2/*newlines*/ ? subject : tail.Substring(0, indexOfEndOfBody).TrimEnd();
+            }
```

Comment:
I propose we extract the functionality as a method that we can test.

I think the method could have the following signature:
```cs
private (string body, string additionalData) ParseCommitBody(...)
```

The check `if (indexOfEndOfBody < 0)` will return `null` and we won't need to duplicate the code for `if (body == null)` check.

Reply:
Agreed.
What do you think about the optimization using `subject.Length + 2/*newlines*/` (which must be done even twice)?

Reply:
I suppose we could use a temp var, but I don't have particularly strong feelings about it

Reply:
My point was: Only the length is checked not the contents of body.

Reply:
Done, factored out the function and added tests.